File: VB/CodeExamples/BasicActions.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.XtraRichEdit
Imports System.Diagnostics
Imports DevExpress.XtraRichEdit.Services
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit.Export

Namespace RichEditDocumentServerAPIExample.CodeExamples

    Public Module BasicActions

        Private Sub CreateNewDocument(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#CreateDocument"
            wordProcessor.CreateNewDocument()
#End Region  ' #CreateDocument
        End Sub

        Private Sub LoadDocument(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#LoadDocument"
            wordProcessor.LoadDocument("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
#End Region  ' #LoadDocument
        End Sub

        Private Sub MergeDocuments(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#MergeDocuments"
            wordProcessor.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            wordProcessor.Document.AppendDocumentContent("Documents//MovieRentals.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
#End Region  ' #MergeDocuments
        End Sub

        Private Sub SplitDocument(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#SplitDocument"
            wordProcessor.LoadDocument("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            'Split a document per page
            Dim pageCount As Integer = wordProcessor.DocumentLayout.GetPageCount()
            For i As Integer = 0 To pageCount - 1
                Dim layoutPage As DevExpress.XtraRichEdit.API.Layout.LayoutPage = wordProcessor.DocumentLayout.GetPage(i)
                Dim mainBodyRange As DevExpress.XtraRichEdit.API.Native.DocumentRange = wordProcessor.Document.CreateRange(layoutPage.MainContentRange.Start, layoutPage.MainContentRange.Length)
                Using tempServer As DevExpress.XtraRichEdit.RichEditDocumentServer = New DevExpress.XtraRichEdit.RichEditDocumentServer()
                    tempServer.Document.AppendDocumentContent(mainBodyRange)
                    'Delete last empty paragraph
                    tempServer.Document.Delete(System.Linq.Enumerable.First(Of DevExpress.XtraRichEdit.API.Native.Paragraph)(tempServer.Document.Paragraphs).Range)
                    'Save the result
                    Dim fileName As String = System.[String].Format("doc{0}.rtf", i)
                    tempServer.SaveDocument(fileName, DevExpress.XtraRichEdit.DocumentFormat.Rtf)
                End Using
            Next

            System.Diagnostics.Process.Start("explorer.exe", "/select," & "doc0.rtf")
#End Region  ' #SplitDocument
        End Sub

        Private Sub SaveDocument(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#SaveDocument"
            wordProcessor.Document.AppendDocumentContent("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            wordProcessor.SaveDocument("SavedDocument.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            System.Diagnostics.Process.Start("explorer.exe", "/select," & "SavedDocument.docx")
#End Region  ' #SaveDocument
        End Sub

        Private Sub PrintDocument(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#PrintDocument"
            wordProcessor.Document.AppendDocumentContent("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            wordProcessor.Print()
#End Region  ' #PrintDocument
        End Sub
    End Module
End Namespace

File: CS/CodeExamples/BasicActions.cs
using System;
using System.Linq;
using DevExpress.XtraRichEdit;

namespace RichEditDocumentServerAPIExample.CodeExamples
{
   public static class BasicActions {

        public static Action<RichEditDocumentServer> CreateNewDocumentAction = CreateNewDocument;
        public static Action<RichEditDocumentServer> LoadDocumentAction = LoadDocument;
        public static Action<RichEditDocumentServer> MergeDocumentsAction = MergeDocuments;
        public static Action<RichEditDocumentServer> SplitDocumentAction = SplitDocument;
        public static Action<RichEditDocumentServer> SaveDocumentAction = SaveDocument;
        public static Action<RichEditDocumentServer> PrintDocumentAction = PrintDocument;
  
        static void CreateNewDocument(RichEditDocumentServer wordProcessor)
        {
            #region #CreateDocument
            // Create a new blank document.
            wordProcessor.CreateNewDocument();
            #endregion #CreateDocument
        }
        static void LoadDocument(RichEditDocumentServer wordProcessor)
        {
            #region #LoadDocument
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);
            #endregion #LoadDocument
        }
        static void MergeDocuments(RichEditDocumentServer wordProcessor)
        {
            #region #MergeDocuments
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents//Grimm.docx", DocumentFormat.OpenXml);

            // Insert content from the file at the document end.
            wordProcessor.Document.AppendDocumentContent("Documents//MovieRentals.docx",DocumentFormat.OpenXml);
            #endregion #MergeDocuments
        }
        static void SplitDocument(RichEditDocumentServer wordProcessor)
        {
            #region #SplitDocument
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);

            // Obtain a number of pages in the document.
            int pageCount = wordProcessor.DocumentLayout.GetPageCount();
            
            // Check all pages in the document.
            for (int i = 0; i < pageCount; i++)
            {
                // Access the document page.  
                DevExpress.XtraRichEdit.API.Layout.LayoutPage layoutPage = wordProcessor.DocumentLayout.GetPage(i);

                // Access the range of the page's main area.
                DevExpress.XtraRichEdit.API.Native.DocumentRange mainBodyRange = wordProcessor.Document.CreateRange(layoutPage.MainContentRange.Start, layoutPage.MainContentRange.Length);

                // Create the temporary RichEditDocumentServer instance.
                using (RichEditDocumentServer tempWordProcessor = new RichEditDocumentServer())
                {
                    // Insert the page content to the instance.
                    tempWordProcessor.Document.AppendDocumentContent(mainBodyRange);
                    // Delete the first empty paragraph.
                    tempWordProcessor.Document.Delete(tempWordProcessor.Document.Paragraphs.First().Range);
                    // Save the document page as an RTF file.
                    string fileName = String.Format("doc{0}.rtf", i);
                    tempWordProcessor.SaveDocument(fileName, DocumentFormat.Rtf);
                }                
            }
            // Open the File Explorer and select the saved file.
            System.Diagnostics.Process.Start("explorer.exe", "/select," + "doc0.rtf");
            #endregion #SplitDocument
        }
        static void SaveDocument(RichEditDocumentServer wordProcessor)
        {
            #region #SaveDocument
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);

            // Save the document as a DOCX file.
            wordProcessor.SaveDocument("SavedDocument.docx", DocumentFormat.OpenXml);
            
            // Open the File Explorer and select the saved file.
            System.Diagnostics.Process.Start("explorer.exe", "/select," + "SavedDocument.docx");
            #endregion #SaveDocument
        }
        static void PrintDocument(RichEditDocumentServer wordProcessor)
        {
            #region #PrintDocument
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);
            
            // Print the document to the default printer with the default settings.
            wordProcessor.Print();
            #endregion #PrintDocument
        }
    }
}
